src/twisted/internet/defer.py

```diff
@@ -929,6 +929,46 @@ def main(reactor):
     return coro
 
 
+def deferredCoro(fn):
```

Comment:
Is there a unit test for this?

Maybe it would be better to add `deferredCoro()` in a separate ticket?

An alternative would be to modify `ensureDeferred()` such that it can operate on coroutine functions in addition to coroutine objects. I'm not sure which approach is better.

Reply:
I've moved deferredCoro to a private `twisted.trial.util_deferredCoro`